src/service/referral.service.ts

```diff
@@ -55,6 +56,28 @@ class ReferralService {
 
     async createReferral(createReferralDto: CreateReferralDto): Promise<Referral> {
         const { name, email, experience, address, roleId, phone, openingId, referredById, resume } = createReferralDto;
+
+        
+        const referrals = await this.referralRepository.findReferralsByEmail(email)
+        let referralWithSameRole;
```

Comment:
If you could move the declaration of referralWithSameRole to where it is initialised and used, it would be more readable. If there is any purpose of keeping it here, then no issues.